Source: http://marc.info/?l=haproxy&m=144476773031348&w=2
Upstream: yes

doc/haproxy-{en,fr}.txt have been removed recently but they were still
referenced in the Makefile. Many other documents have also been
added. Instead of hard-coding a list of documents to install, install
all those in doc/ with some exceptions:

 - coding-style.txt is more for developers
 - gpl.txt and lgpl.txt are usually present at other places (and I would
   have to remove them in the Debian packaging, less work for me)

The documentation in the subdirectories is not installed as it is more
targeted to developers.
---
 Makefile | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index a7da80b56fa0..4f0aa8af705b 100644
--- a/Makefile
+++ b/Makefile
@@ -796,9 +796,12 @@ install-man:
 	install -d "$(DESTDIR)$(MANDIR)"/man1
 	install -m 644 doc/haproxy.1 "$(DESTDIR)$(MANDIR)"/man1
 
+EXCLUDE_DOCUMENTATION = lgpl gpl coding-style
+DOCUMENTATION = $(filter-out $(EXCLUDE_DOCUMENTATION),$(patsubst doc/%.txt,%,$(wildcard doc/*.txt)))
+
 install-doc:
 	install -d "$(DESTDIR)$(DOCDIR)"
-	for x in configuration architecture haproxy-en haproxy-fr; do \
+	for x in $(DOCUMENTATION); do \
 		install -m 644 doc/$$x.txt "$(DESTDIR)$(DOCDIR)" ; \
 	done
 
@@ -811,7 +814,7 @@ install: install-bin install-man install-doc
 
 uninstall:
 	rm -f "$(DESTDIR)$(MANDIR)"/man1/haproxy.1
-	for x in configuration architecture haproxy-en haproxy-fr; do \
+	for x in $(DOCUMENTATION); do \
 		rm -f "$(DESTDIR)$(DOCDIR)"/$$x.txt ; \
 	done
 	-rmdir "$(DESTDIR)$(DOCDIR)"
